### PR TITLE
fix(tv-mode): use canonical dashboard origin

### PR DESCRIPTION
**Pull request by AddarshKS**
## Summary

Use the existing dashboard URL configuration consistently for TV Mode instead of requiring a separate TV-specific origin setting.

- Resolve the TV API origin from `NEXT_PUBLIC_HEXCLAVE_DASHBOARD_URL`, preserving the existing `NEXT_PUBLIC_STACK_DASHBOARD_URL` compatibility alias.
- Align dashboard TV links with that same setting.
- Remove the obsolete `HEXCLAVE_TV_DISPLAY_ORIGIN` environment example and its override precedence.
- Add regression coverage for canonical configuration, legacy aliases, stale overrides, cookie policy, and both TV API route aliases.

Pairing, credential lifecycle, database schema, and appliance behavior are unchanged. No new environment variable, dependency, or migration is required.

## Validation

- 43 focused backend/dashboard tests passed.
- Focused ESLint, backend and E2E typechecks, and diff whitespace checks passed.
- Added a live API preflight test; execution is pending a running development backend.
- Full dashboard typechecking is blocked by generated `.next` references to unrelated missing GTM pages.

## Deployment

Use the existing canonical dashboard URL setting in each environment. A deployed backend must pick up this change before the updated origin selection takes effect.

---
## Summary by cubic
TV Mode now derives its API origin and dashboard links from canonical `NEXT_PUBLIC_HEXCLAVE_DASHBOARD_URL`, keeping `NEXT_PUBLIC_STACK_DASHBOARD_URL` as a compatibility alias instead of TV-specific or browser-specific overrides. This prevents stale origin settings from changing credentialed CORS or cookie behavior.

**Bug Fixes**

- Covers credentialed CORS for both `/api/latest` and `/api/v1` TV routes.
- Preserves strict cookie behavior for same-site and development displays, with `SameSite=None` only for secure cross-origin displays.
- Throws when the canonical and legacy dashboard settings contain conflicting values.

**Migration**

- Remove `HEXCLAVE_TV_DISPLAY_ORIGIN` from deployments and configure the existing canonical dashboard URL instead.
- Restart the backend after changing the dashboard URL; the origin is captured once at startup.

<sup>Written for commit d95a4b68d9519a8df693d07d5604ff27cdf3a5dd. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/hexclave/hexclave/pull/2065?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved TV display URL and origin detection using the canonical dashboard configuration.
  - Ensured credentialed TV-display CORS requests are allowed only from the configured dashboard origin.
  - Added support for matching legacy dashboard settings while rejecting conflicting values.
  - Improved handling of invalid, missing, stale, and untrusted origins.

- **Tests**
  - Expanded coverage for TV display URLs, CORS behavior, cookie security, route aliases, and configuration edge cases.

### CODE REVIEW

File: apps/e2e/tests/backend/endpoints/api/v1/tv-display-cors.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { it, niceFetch, STACK_BACKEND_BASE_URL, STACK_DASHBOARD_BASE_URL } from "../../../../helpers";
+
+it("permits credentialed TV preflights only from the configured dashboard on both route aliases", async ({ expect }) => {
+  const dashboardOrigin = new URL(STACK_DASHBOARD_BASE_URL).origin;
```
[the remark this answers is not in the document]
Comment (AddarshKS) [reply apps/e2e/tests/backend/endpoints/api/v1/tv-display-cors.test.ts:4]: This helper represents the E2E dashboard address and already supports HEXCLAVE_DASHBOARD_BASE_URL through the shared environment resolver. Keeping it independent of the backend configuration is intentional: the E2E test verifies externally observable CORS behavior. Canonical/legacy origin selection and stale overrides are covered separately in display-origin.test.ts. Reading an optional backend environment variable in the test process would not reliably verify the separately running backend’s configuration, so we’re retaining this helper.

**Comment by AddarshKS** [comment]
@coderabbitai Regarding the [post-build resolver caution](https://github.com/hexclave/hexclave/pull/2065#pullrequestreview-5215073215):

The supported post-build startup already handles this through `mirror_hexclave_stack_env()` in `docker/server/entrypoint.sh`. It runs before sentinel substitution and mirrors canonical-only configuration into the legacy alias, so `STACK_ENV_VAR_SENTINEL_NEXT_PUBLIC_STACK_DASHBOARD_URL` resolves without requiring the operator to supply the legacy variable. The dashboard URL is also a required sentinel, and startup checks for incomplete replacement. We verified canonical-only and legacy-only inputs; no resolver change is necessary.

**Review by BilalG1 — APPROVED** [review]
